src/main/java/com/internet/shop/controller/RegistrationController.java

```diff
@@ -0,0 +1,52 @@
+package com.internet.shop.controller;
+
+import com.internet.shop.library.Injector;
+import com.internet.shop.model.ShoppingCart;
+import com.internet.shop.model.User;
+import com.internet.shop.service.interfaces.ShoppingCartService;
+import com.internet.shop.service.interfaces.UserService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class RegistrationController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private static UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+    private static ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/registration.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String name = req.getParameter("name");
+        String login = req.getParameter("login");
+        String password = req.getParameter("password");
+        String passwordRepeat = req.getParameter("password-repeat");
+
+        if (name.length() == 0
+                || login.length() == 0
+                || password.length() == 0) {
```

Comment:
we can make this checking at the jsp page, using HTML tags 
Input Type Range -> https://www.w3schools.com/html/html_form_input_types.asp

Reply:
But that is not safe, in my opinion
Because the 'required' part can optionally be deleted by the user
Should I change it?

![photo_2020-09-09_21-35-21](https://user-images.githubusercontent.com/68664175/92696429-860ba780-f352-11ea-8b74-2b891fdff34c.jpg)

Reply:
Bohdan allowed not to remove, thanks :)
But I'll add the check to the .jsp file

Reply:
да, я повністю погоджуюсь що в ідеалі варто мати кілька рівнів захисту і валідації даних. 
Проте, якщо уже й валідувати дані, то варто робити перевірку на null,  потім перевірку на пустоту строки. Для пароля можна перевіряти щоб довжина була не менша 8-10 симолів і щоб були як цифри так і букви. А для логіна - свою валідацію робити на наявність @ і відсутність спец символів і т.д
А це все - нагромадження коду... і контроллер виглядатиме не ок.
Зовсім скоро ми вивчимо що таке сервлетні фільтри. Можна буде додати в окремий фільтр валідацію за цими всіма парметрами) Так в кожного класу буде своя зона відповідальності: фільтр - валідує, контроллер - обробляє запит, надає відповідь